src/streamlink/plugins/wasd.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.stream import HLSStream
+
+log = logging.getLogger(__name__)
+
+class WASD(Plugin):
+	_url_re = re.compile(r"https?://(?:www\.)?wasd\.tv/")
+	_url_channel = re.compile(r"https?://(?:www\.)?wasd\.tv/channel/(\d+)")
+	_url_video = re.compile(r"https?://(?:www\.)?wasd\.tv/channel/\d+/videos/(\d+)")
+	
+	@classmethod
+	def can_handle_url(cls, url):
+		return WASD._url_re.match(url)
+		
+	def _get_streams(self):
+		re_video_id = self._url_video.match(self.url)
+		re_channel_id = self._url_channel.match(self.url)
+		
+		if not re_video_id and not re_channel_id:
+			return log.error("Expected to see url like https://wasd.tv/channel/12345 or https://wasd.tv/channel/12345/videos/67890")
+		
+		# Check if we have AUTH token stored in session cache and request it
+		cookies = self.session.http.cookies
+		if not cookies or not cookies['cronos-auth-token'] or not cookies['cronos-auth-token-signature']:
+			log.debug("There is no AUTH token in cache. Attempting to request one")
+			self.getAuthToken()
+		if not cookies or not cookies['cronos-auth-token'] or not cookies['cronos-auth-token-signature']:
+			return log.error("Failed to get AUTH token")
+		
+		if re_video_id:
+			streams = self.getStreamsFromURL("https://wasd.tv/api/media-containers/" + re_video_id.group(1), 'video')
+		else:
+			streams = self.getStreamsFromURL("https://wasd.tv/api/media-containers?media_container_status=RUNNING&limit=1&offset=0&channel_id=" + re_channel_id.group(1) + "&media_container_type=SINGLE,COOP", 'channel')
+		
+		if streams:
+			for video in streams:
+				# TODO: This is how I planned it initially. video[0] is quality and video[1] is URL for .m3u8 file. 
```

Comment:
no need for such long comments and `video[0]` is not needed if you use 
`HLSStream.parse_variant_playlist`